ens/main.py

```diff
@@ -282,6 +282,54 @@ def owner(self, name: str) -> ChecksumAddress:
         node = raw_name_to_hash(name)
         return self.ens.caller.owner(node)
 
+    def get_text(self, name: str, key: str) -> str:
+        """
+        Get the value of a text record by key from an ENS name.
+
+        :param str name: ENS name to look up
+        :param str key: ENS name's text record key
+        :return:  ENS name's text record value
+        :rtype: str
+        """
+        node = raw_name_to_hash(name)
+
+        r = self.resolver(name)
+        if r:
+            return r.caller.text(node, key)
+        else:
+            raise UnownedName("claim domain using setup_address() first")
+
+    def set_text(
+        self,
+        name: str,
+        key: str,
+        value: str,
+        transact: "TxParams" = {}
+    ) -> HexBytes:
+        """
+        Set the value of a text record of an ENS name.
+
+        :param str name: ENS name
+        :param str key: Name of the attribute to set
+        :param str value: Value to set the attribute to
+        :param dict transact: the transaction configuration, like in
+            :meth:`~web3.eth.Eth.send_transaction`
+        :return: Transaction hash
+        :rtype: HexBytes
+        :raises UnownedName: if no one owns `name`
+        """
+        owner = self.owner(name)
+        node = raw_name_to_hash(name)
+
+        if transact == {}:
+            transact['from'] = owner
```

Comment:
Note to self: I think I actually want to do a merge here, so that if a user passes in a transaction dict without a `from` key, it will also default to `owner`